File: java/Three/Three/Laws.java
package Three.Three;
public abstract class Laws {
    protected String name;
    protected String explanation;

    public Laws(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public String getExplanation() {
        return explanation;
    }

    public abstract float calculateLaw();
}

File: Three/Laws.php

<?php
abstract class Laws {
    protected string $name;
    protected string $explanation;

    public function __construct(string $name){
        $this->name = $name;
    }
    public function getName(): string {
        return $this->name;
    }
    public function getExplanation(): string {
        return $this->explanation;
    }
    abstract public function calculateLaw(): float;
}
?>